File: baek_1697.py
import sys
from collections import deque

def bfs(a,b):
    q=deque()
    vi=[False]*100001
    q.appendleft(a)
    vi[a]=True
    x=1
    y=0
    cnt=0
    while len(q)!=0:
        cur=q.pop()
        if cur==b:
            print(cnt)
            return
        if cur!=0 and not vi[cur-1]:
            vi[cur-1]=True
            y+=1
            q.appendleft(cur-1)
        if cur!=100000 and not vi[cur+1]:
            vi[cur+1]=True
            y+=1
            q.appendleft(cur+1)
        if cur<=50000 and not vi[cur*2]:
            vi[cur*2]=True
            y+=1
            q.appendleft(cur*2)
        x-=1
        if x==0:
            cnt+=1
            x,y=y,x

n,k=map(int,sys.stdin.readline().strip().split())
bfs(n,k)

File: baek_1697.cpp
#include <stdio.h>
#include <queue>
using namespace std;

void bfs(int n, int k){
    int ctr[2] = {1,0};
    int cnt = 0;
    queue<int> q;
    bool vi[200002] = { false, };
    int cur;
    q.push(n);
    vi[n] = true;
    while(!q.empty()){
        cur = q.front();
        q.pop();
        if(cur==k){
            printf("%d\n", cnt);
            return;
        }
        if(cur!=100000 && !vi[cur+1]){
            vi[cur+1] = true;
            ctr[(cnt+1)%2]++;
            q.push(cur+1);
        }
        if(cur!=0 && !vi[cur-1]){
            vi[cur-1] = true;
            ctr[(cnt+1)%2]++;
            q.push(cur-1);
        }
        if(cur<= 50000 && !vi[cur*2]){
            vi[cur*2] = true;
            ctr[(cnt+1)%2]++;
            q.push(cur*2);
        }
        ctr[cnt%2]--;
        if(ctr[cnt%2]==0)
            cnt++;
    }
}

int main(void){
    int n, k;
    scanf("%d %d", &n, &k);
    bfs(n,k);
    return 0;
}